### PR TITLE
 db: Fix database operators assignment

### PR DESCRIPTION
**Pull request by panosfol**

### CODE REVIEW

File: src/db.c [hunk 1]
```diff
@@ -31,18 +32,20 @@ int execute_db_operations(void)
 			free(available_dbs);
 			return ret;
 		}
+
+		if (db_ops_counter + 1 < MAX_AVAILABLE_DBS) {
+			db_ops_counter++;
+		}
```
Comment (charmitro) [inline src/db.c:38]: Remove, it is too late in the scope to do this.

File: src/postgres.c [hunk 2]
```diff
@@ -31,26 +32,24 @@ static struct db_operations pg_db_ops = {
...
 
 	pg_db_t->pg_conf = CNC_MALLOC(sizeof(postgres_t));
 	memcpy(pg_db_t->pg_conf, ini_config->postgres_config,
 	       sizeof(postgres_t));
 
 	pg_db_ops.db = pg_db_t;
-	available_dbs[0] = &pg_db_ops;
+	available_dbs[db_ops_counter] = &pg_db_ops;
 
 	return 0;
```
Comment (charmitro) [inline src/postgres.c:50]: I'd like to see something like:
```c
	if (db_ops_counter < MAX_AVAILABLE_DBS) {
		struct db_t *pg_db_t = CNC_MALLOC(sizeof(struct db_t));

		pg_db_t->pg_conf = CNC_MALLOC(sizeof(postgres_t));
		memcpy(pg_db_t->pg_conf, ini_config->postgres_config,
		       sizeof(postgres_t));

		pg_db_ops.db = pg_db_t;
		available_dbs[0] = &pg_db_ops;
		available_dbs[db_ops_counter++] = &pg_db_ops;
		return 0;
	} else {
		return -2; // Out of bounds.
	}
```

**Review by charmitro — CHANGES_REQUESTED** [review]

File: src/db.c (hunk 1 above)
Comment (panosfol) [reply src/db.c:38]: I believe It shouldnt be the responsibility of the init function to incriment the counter.
The counter is created and should be managed by the `db` file no?

Also its better for designing the new database implementation to not have to think about managing the 
counter by incrementing it and checking if its out of bounds, i believe this is something that should be done as a different procedure and not in every `init_func` for each database system we implement

Comment (charmitro) [reply src/db.c:38]: A better way to do it is:
1. Check if counter is out of bounds
2. If not, run init function
3. If it is, exit

Comment (panosfol) [reply src/db.c:38]: Ok sure. Where should the increment happen though? We could do it after successfully running the init function.

Comment (charmitro) [reply src/db.c:38]: Yes, run init -> increment

File: src/postgres.c (hunk 2 above)
Comment (charmitro) [reply src/postgres.c:50]: Resolved by https://github.com/terra-sync/cnc/pull/64#discussion_r1561506395

**Review by charmitro — APPROVED** [review]

**Pull request merged**